server/routing.go

```diff
@@ -0,0 +1,80 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Layr-Labs/eigenda-proxy/commitments"
+	"github.com/gorilla/mux"
+)
+
+func (svr *Server) registerRoutes(r *mux.Router) {
+	subrouterGET := r.Methods("GET").PathPrefix("/get").Subrouter()
+	// simple commitments (for nitro)
+	subrouterGET.HandleFunc("/"+
+		"{optional_prefix:(?:0x)?}"+ // commitments can be prefixed with 0x
+		"{version_byte_hex:[0-9a-fA-F]{2}}"+ // should always be 0x00 for now but we let others through to return a 404
+		"{raw_commitment_hex:[0-9a-fA-F]*}",
+		withLogging(withMetrics(svr.handleGetSimpleCommitment, svr.m, commitments.SimpleCommitmentMode), svr.log),
```

Comment:
what benefits do we reap by using the gorilla library? I.e, why use a heavy framework vs. what we had before. Generally agree with the structure of the refactoring but still a bit curious if we need to introduce a heavy framework with more developer complexity.

Reply:
also from a philosophical standpoint I've never seen ethereum projects use third party frameworks like this in RPC server implementations

Reply:
Oh I've got many mouthfuls to say about this haha:
1. re ethereum projects not using web frameworks: not quite sure I haven't done an extensive survey. A lot of them are simple json-rpc, so they use a json-rpc library instead of a rest api framework, but the altda decided to go with rest api. Also if I had to guess I'd say if other projects that are (proxy) servers like ours (like flashbots builder say) are not using web framework either out of ignorance, or because their routing is super simple. But our routing is a bit more involved, so basic stdlib http router (even after go1.22 improvements) isn't enough.
2. gorilla is not a heavy framework. it's actually extremely lightweight. If I had my way I would actually use a full fledged framework with middlewares, like gin or echo (note that even these are "lightweight" frameworks). But gorilla mux is purely a routing library, not a framework. See limitations section at the bottom of [this](https://eli.thegreenplace.net/2021/rest-servers-in-go-part-3-using-a-web-framework/#Limitations) for a comparison.
3. I already tried to explain the benefits of using gorilla's router instead of stdlib's router. I think this [article](https://matklad.github.io/2023/11/15/push-ifs-up-and-fors-down.html) summarizes my thoughts regarding the problem with our current implementation better than I ever could. Basically our routing ifs/switches are sprinkled everywhere in the proxy. This makes for very difficult reading. There's really only 3 types of commitments, so we should make the choice one of which "row" we're in, at the top routing layer, and then the rest of the code should be aware of which route it's in and be able to specialize. See for eg this [switch statement](https://github.com/Layr-Labs/eigenda-proxy/blob/refactor--better-routing-using-gorilla-mux-patterns/store/router.go#L123); don't think we should have this that far down code paths.